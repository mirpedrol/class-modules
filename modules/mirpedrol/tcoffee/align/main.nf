process TCOFFEE_ALIGN {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-a76a981c07359a31ff55b9dc13bd3da5ce1909c1:84c8f17f1259b49e2f7783b95b7a89c6f2cb199e-0':
        'biocontainers/mulled-v2-a76a981c07359a31ff55b9dc13bd3da5ce1909c1:84c8f17f1259b49e2f7783b95b7a89c6f2cb199e-0' }"

    input:
    tuple val(meta) ,  path(fasta)

    output:
    tuple val(meta), path("*.aln.gz"), emit: alignment
    path "versions.yml"              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    export TEMP='./'
    t_coffee -seq ${fasta} \
        $args \
        -output fasta_aln \
        -thread ${task.cpus} \
        -outfile stdout \
        | pigz -cp ${task.cpus} > ${prefix}.aln.gz

    # If stdout file exist, then compress the file
    # This is a patch for the current behaviour of the regressive algorithm
    # that does not support the stdout redirection
    if [ -f stdout ]; then
        pigz -cp ${task.cpus} < stdout > ${prefix}.aln.gz
        rm stdout
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tcoffee: \$( t_coffee -version | awk '{gsub("Version_", ""); print \$3}')
        pigz: \$(echo \$(pigz --version 2>&1) | sed 's/^.*pigz\\w*//' ))
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.aln.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tcoffee: \$( t_coffee -version | awk '{gsub("Version_", ""); print \$3}')
        pigz: \$(echo \$(pigz --version 2>&1) | sed 's/^.*pigz\\w*//' ))
    END_VERSIONS
    """
}
