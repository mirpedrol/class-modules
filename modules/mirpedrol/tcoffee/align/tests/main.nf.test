nextflow_process {

    name "Test Process TCOFFEE_ALIGN"
    script "../main.nf"
    process "TCOFFEE_ALIGN"

    tag "modules"
    tag "modules_mirpedrol"
    tag "tcoffee"
    tag "tcoffee/align"
    tag "untar"

    test("fasta - align_sequence") {

        when {
            process {
                """
                input[0] = [ [ id:'test' ],
                                file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                            ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.alignment).match("alignment")},
                { assert snapshot(process.out.versions).match("versions") }
            )
        }
    }
}