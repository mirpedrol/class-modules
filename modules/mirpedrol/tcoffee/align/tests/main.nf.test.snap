{
    "versions": {
        "content": [
            [
                "versions.yml:md5,fb187c9186b50a8076d08cd3be3c1b70"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-10T15:28:18.280597"
    },
    "alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,bd1db08ad04514cc6d1334598c1a6ef0"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-02-28T18:59:35.169119"
    }
}